### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by romanisb**
Biggest change is to use the Jenkins plugin BOM which simplifies the dependency management. Everything else remains untouched except some test resources. Here the message output changed a little with the newer Jenkins core and plugin versions.

### CODE REVIEW

**Review by rinckm — APPROVED** [review]
lgtm

File: pom.xml [hunk 1]
```diff
@@ -6,19 +6,26 @@
...
     <artifactId>pipeline-elasticsearch-logs</artifactId>
-    <version>0.10.0</version>
+    <version>0.11.0</version>
     <packaging>hpi</packaging>
     <properties>
-        <jenkins.version>2.176.4</jenkins.version>
-        <java.level>8</java.level>
+        <!-- Adapt the jenkins.tools.bom artifact ID in the dependencyManagement section
+             if the Jenkins core version is changed -->
+        <jenkins.version>2.319.1</jenkins.version>
```
Comment (alxsap) [inline pom.xml:19]:
```suggestion
        <jenkins.version>2.332.3</jenkins.version>
```

**Comment by alxsap** [comment]
it seems guava was removed between jenkins-core `2.319.1` and `2.332.3`. This also means at runtime guava will not be available anymore and older plugin versions will fail!

```
15:26:52  [INFO] -------------------------------------------------------------
15:26:52  [ERROR] COMPILATION ERROR : 
15:26:52  [INFO] -------------------------------------------------------------
15:26:52  [ERROR] /home/jenkins/agent/workspace/steward-ci-os/elasticsearch-logs-plugin-pipeline/src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/direct_es/ElasticSearchWriteAccessDirect.java:[3,40] cannot find symbol
15:26:52    symbol:   class Ranges
15:26:52    location: package com.google.common.collect
15:26:52  [ERROR] /home/jenkins/agent/workspace/steward-ci-os/elasticsearch-logs-plugin-pipeline/src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/direct_es/ElasticSearchWriteAccessDirect.java:[3,1] static import only from classes and interfaces
```

**Review by alxsap — DISMISSED** [review]
Looks good, tests pending...
see https://github.com/SAP/stewardci-jenkinsfilerunner-image/pull/90

File: pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <properties>
         <!-- Adapt the jenkins.tools.bom artifact ID in the dependencyManagement section
              if the Jenkins core version is changed -->
-        <jenkins.version>2.319.1</jenkins.version>
+        <jenkins.version>2.332.3</jenkins.version>
```
Comment (romanisb) [inline pom.xml:19]: If the Jenkins version is changed, the Jenkins bom artifact should be adapted accordingly as well.

File: pom.xml [hunk 3]
```diff
@@ -124,86 +122,27 @@
...
-                <artifactId>ssh-credentials</artifactId>
-                <version>1.17.3</version>
-            </dependency>
-            <dependency>
-                <groupId>org.apache.commons</groupId>
-                <artifactId>commons-lang3</artifactId>
-                <version>3.7</version>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-2.319.x</artifactId>
+                <version>${bom-2.319.x.version}</version>
```
Comment (alxsap) [inline pom.xml:137]:
```suggestion
                <artifactId>bom-2.332.x</artifactId>
                <version>${bom-2.332.x.version}</version>
```

File: pom.xml [hunk 4]
```diff
@@ -6,19 +6,26 @@
...
         <useBeta>true</useBeta>
-        <workflow-api-plugin.version>2.40</workflow-api-plugin.version>
-        <fluency.version>2.4.1</fluency.version>
+
+        <mock-slave.version>109.vac7845f10470</mock-slave.version>
+        <unique-id.version>2.2.1</unique-id.version>
+        <fluency.version>2.6.4</fluency.version>
+        <jsr305.version>3.0.2</jsr305.version>
+
+        <bom-2.319.x.version>1342.v729ca_3818e88</bom-2.319.x.version>
```
Comment (alxsap) [inline pom.xml:27]:
```suggestion
        <bom-2.332.x.version>1409.v7659b_c072f18</bom-2.332.x.version>
```

File: pom.xml (hunk 2 above)
Comment (romanisb) [reply pom.xml:19]: Yes, exactly. I tried to reflect this in the comment above. Maybe we should rewrite it accordingly to improve the intention behind it.

File: pom.xml [hunk 5]
```diff
@@ -6,19 +6,26 @@
...
     <groupId>io.jenkins.plugins</groupId>
     <artifactId>pipeline-elasticsearch-logs</artifactId>
-    <version>0.10.0</version>
+    <version>0.11.0</version>
     <packaging>hpi</packaging>
     <properties>
-        <jenkins.version>2.176.4</jenkins.version>
-        <java.level>8</java.level>
+        <!-- Adapt the jenkins.tools.bom artifact ID in the dependencyManagement section
+             if the Jenkins core version is changed -->
```
Comment (alxsap) [inline pom.xml:18]:
```suggestion
        <!-- If the "jenkins.version" is changed you also need to adjust:
            - the major and minor version in the "<bom-#.###.x.version>" tag below (keep "x" as patch version)
            - in the dependencyManagement section the "io.jenkins.tools.bom:bom-#.###.x" artifactId and version property name.
        -->
```

File: pom.xml (hunk 2 above)
Comment (alxsap) [reply pom.xml:22]: @romanisb  OK now?

File: pom.xml (hunk 3 above)
Comment (alxsap) [reply pom.xml:141]: Update bom artifact

File: pom.xml (hunk 2 above)
Comment (romanisb) [reply pom.xml:22]: Looks good.

**Review by alxsap — APPROVED** [review]

**Pull request merged**